### ISSUE TITLE
This won't work for 10.6 PowerPC, right?

### ISSUE THREAD

**Issue by barracuda156** [mistydemeo/tigerbrew#775]
I am looking for alternative of MacPorts, which does build on 10.6 PPC, but some ports fail to build.
I am specifically talking about 10.6 PowerPC, not Intel. I have 10A190 build installed and running.

**Comment by mistydemeo** [comment]
10.6 PowerPC? I had no idea it existed! It hasn't been tested but I'd absolutely welcome any reports, and try to fix anything that's broken.

**Comment by LukeUsher** [comment]
> 10A190

There's no retail/final 10.6, 10A190 is a beta from before they removed PowerPC support, that was handed out at some Apple Developer Conference events.

**Comment by Maltettt** [comment]
I‘ve tried using Tigerbrew on the current 10.6 PPC build (Alpha Testing: Version 5), but brew tries to install package versions for the wrong CPU type (Bad CPU type in executable) for software that works fine on 10.5 PPC.

**Comment by sevan** [comment]
What command did you run & which package was it installing?

**Comment by sevan** [comment]
I wonder if commenting out line [28 & 29](https://github.com/mistydemeo/tigerbrew/blob/9216073492b151fd87d111adb410bf2dbaf47d72/Library/Homebrew/bottles.rb#L28C1-L29C64) in `Library/Homebrew/bottles.rb` will prevent it from making the wrong selection.

**Comment by Maltettt** [comment]
I’ve tried of running brew doctor. Just now I‘ve realised, that the bad executable is actually clang. I‘ve tried running 10.6 PPC with XCode 3.2.6 and 3.1.4 and had no luck correcting that issue for now.

**Comment by barracuda156** [comment]
> I’ve tried of running brew doctor. Just now I‘ve realised, that the bad executable is actually clang. I‘ve tried running 10.6 PPC with XCode 3.2.6 and 3.1.4 and had no luck correcting that issue for now.

1. Please do not use Leopard Xcode on 10.6.x, do yourself a favor.
2. Xcode 3.2.6 needs a number of executables rebuilt/replaced/substituted, because not everything has ppc slices. The easiest way which is acceptable is to pull over stuff from 10a190 installed on a drive or partition. Not advised, you get poor man’s Xcode. The proper alternatives are either to build from Apple sources (manually or via darwinbuild) or install something newer which works for the purpose (you only need a handful of CLI tools).
3. Clang is broken anyway, just delete it (or rename to clang_broken). There is absolutely no value in having it, it only creates troubles.
4. For the record, A5 image has a wrong bsdtar (it is renamed gnutar from 10a190). Delete it and pick a correct one from official 10.6.8, it has ppc slice and works fine.

Myself I install darwin-xtools into /usr/bin, build make from Apple source and the rest use from ports (no real need to bother with archaic bison, m4 and flex).
See macos-powerpc.org for some details. (Sorry for referring to myself LOL, but I don’t know a better solution.)

P. S. Just in case, you don’t need to trust my installers, everything can be bootstrapped.

**Comment by mistydemeo** [comment]
I was about to say the bottle tags might be wrong, but checking the code I think it's fine: https://github.com/mistydemeo/tigerbrew/blob/master/Library/Homebrew/bottles.rb#L32-L33

On PPC we include the PowerPC arch as part of the bottle tag, so you won't get just `snow_leopard` (which is Intel). In that case it does sound like the issue just the developer tools. Not sure how we want to handle that... 

Homebrew/Tigerbrew has always had a very specific assumption that we're using a set of Xcode tools, and that the Xcode-installed compilers will be usable. I wouldn't be against making improvements to this stuff for Snow Leopard PPC, but there's definitely a lot to think about for how we want this to work, how we want to be able to bootstrap our way to compatible compilers, etc.

**Comment by barracuda156** [comment]
@mistydemeo A user of 10.6.8 on ppc must do a minimal set-up from his side, since at the very minimum working ar, as, ld and make are required to bootstrap anything (they are needed before MacPorts is installed, for example). However you can implement a bootstrap from vanilla Xcode if you do that on 10.6.8 x86 with Rosetta. Once basic tools are built, those can be installed into 10.6.8 on ppc, and then a user can proceed with a package manager of choice normally.

**Comment by barracuda156** [comment]
> I was about to say the bottle tags might be wrong, but checking the code I think it's fine: https://github.com/mistydemeo/tigerbrew/blob/master/Library/Homebrew/bottles.rb#L32-L33

@mistydemeo 
I am unaware of its context, but the logic there looks wrong to me (and unnecessarily complicated). The query for arch happens only for <= 10.5, and default assumption will be x86; for SL only bitness is checked. Why at all SL needs to be a special case? It does not have ppc64 support, okay, but that can be handled.

**Comment by mistydemeo** [comment]
That Snow Leopard-specific check comes from Homebrew, and predates the Tigerbrew fork even existing. It comes from 32-bit Snow Leopard, which Homebrew very briefly supported with bottles: 041060e28fa0e0c6ed6995ec34bb62188f159d2a

I'll look at updating the bottle check here to add Snow Leopard PPC support.

**Comment by sevan** [comment]
Anyone with an instance of Snow Leopard on PowerPC want to give the patch in https://github.com/mistydemeo/tigerbrew/pull/1495 a try?

**Comment by kristibektashi** [comment]
I would like to note that apart from the MacOS 10.6 beta, there is also a custom distribution of MacOS 10.6.8 PowerPC, which is built from files taken from 10.5, the 10.6 beta, as well as the PowerPC/Rosetta slices of 10.6.8 Intel, and either the kernel from the beta or a recompiled XNU kernel (don't remember which). So that's something to look out for

**Comment by kristibektashi** [comment]
Found more info on the 10.6.8 custom distribution: https://lowendmac.com/2024/mac-os-x-snow-leopard-10-6-10-6-8-for-powerpc/

**Comment by kristibektashi** [comment]
Note that MacintoshGarden has "Alpha Build 5" of 10.6.8 PPC, which is newer than the "Alpha Build 3" mentioned in the article I sent above (the below website is directly linked in the above one)

https://macintoshgarden.org/apps/mac-os-x-1068-powerpc-alpha-build-3-macrumors

**Comment by kristibektashi** [comment]
More info: https://forums.macrumors.com/threads/10-6-snow-leopard-powerpc-development.2439769/

They seem to have fixed the internet connectivity issues in Alpha 5

**Comment by sevan** [comment]
@kristibektashi are you running any build of OS X 10.6 on PowerPC?

**Comment by kristibektashi** [comment]
> [@kristibektashi](https://github.com/kristibektashi) are you running any build of OS X 10.6 on PowerPC?

No, I just stumbled upon this issue and wanted to give out this info that has been living in the back of my head for a while, because this 10.6.8 PPC build is magnificent stuff in my opinion and I didn't want it to be lost to time

**Comment by sevan** [comment]
Ok, need someone to try Misty's patch mentioned earlier here.
